src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisResultBuilder.cs

```diff
@@ -49,6 +53,37 @@ internal AnalysisResultBuilder(bool logAnalyzerExecutionTime, ImmutableArray<Dia
             return map;
         }
 
+        private static ImmutableDictionary<string, OneOrMany<AdditionalText>> CreatePathToAdditionalTextMap(ImmutableArray<AdditionalText> additionalFiles)
+        {
+            if (additionalFiles.IsEmpty)
+            {
+                return ImmutableDictionary<string, OneOrMany<AdditionalText>>.Empty.WithComparers(PathUtilities.Comparer);
```

Comment:
>ImmutableDictionary<string, OneOrMany<AdditionalText>>.Empty.WithComparers(PathUtilities.Comparer); [](start = 23, length = 99)

Consider creating a `static readonly` field for this value, to avoid allocating each time.